File: 99-Continue/164/leftmost.py
from collections import deque

class Node:
    def __init__(self,data:int):
        self.data = data
        self.left = self.right = None

class BinaryTree:
    def __init__(self):
        self.root = None
    
    def buildTree(self,arr:list[str])->Node:
        if len(arr)==0:
            return self.root
        self.root = Node(int(arr[0]))
        queue = deque([self.root])
        i = 1
        while queue and i<len(arr):
            current = queue.popleft()
            if i<len(arr):
                current.left = Node(int(arr[i]))
                queue.append(current.left)
            i+=1
            if i<len(arr):
                current.right = Node(int(arr[i]))
                queue.append(current.right)
            i+=1
        return self.root
        
    def leftMost(self)->int:
        current = self.root
        while current.left is not None:
            current=current.left
        return current.data

def main():
    int(input())
    s = input().split()
    tree = BinaryTree()
    tree.buildTree(s)
    print(tree.leftMost())
    
if __name__=="__main__":
    main()

File: 99-Continue/164/Leftmost.java
import java.util.*;

public class Leftmost {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        String s = sc.nextLine();
        String[] arr = s.split(" ");
        BinaryTree tree = new BinaryTree();
        tree.buildTree(arr);
        System.out.println(tree.leftMost());
        sc.close();
    }
}

class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data = data;
        this.left = this.right = null;
    }
}

class BinaryTree{
    Node root;
    BinaryTree(){
        this.root = null;
    }
    
    Node buildTree(String[] arr){
        if(arr.length==0){
            return root;
        }
        root = new Node(Integer.parseInt(arr[0]));
        Queue<Node> queue = new LinkedList<>();
        queue.offer(root);
        int i=1;
        while(!queue.isEmpty()&&i<arr.length){
            Node current = queue.poll();
            if(i<arr.length && !arr[i].equals("null")){
                current.left = new Node(Integer.parseInt(arr[i]));
                queue.offer(current.left);
            }
            i++;
            if(i<arr.length && !arr[i].equals("null")){
                current.right = new Node(Integer.parseInt(arr[i]));
                queue.offer(current.right);
            }
            i++;
        }
        return root;
    }
    
    int leftMost(){
        if(root==null){
            return 0;
        }
        Node current = root;
        
        while(current.left!=null){
            current=current.left;
        }
        return current.data;
    }
}